diff --git a/source/vendor/supports-color/index.js b/source/vendor/supports-color/index.js
index 13883726..0e130a18 100644
--- a/source/vendor/supports-color/index.js
+++ b/source/vendor/supports-color/index.js
@@ -135,6 +135,10 @@ function _supportsColor(haveStream, {streamIsTTY, sniffFlags = true} = {}) {
 		return 3;
 	}
 
+	if (env.TERM === 'xterm-ghostty') {
+		return 3;
+	}
+
 	if ('TERM_PROGRAM' in env) {
 		const version = Number.parseInt((env.TERM_PROGRAM_VERSION || '').split('.')[0], 10);
 
