diff --git a/src/addBusinessDays/index.ts b/src/addBusinessDays/index.ts
index f771dea4c3..ee1b39578c 100644
--- a/src/addBusinessDays/index.ts
+++ b/src/addBusinessDays/index.ts
@@ -6,7 +6,7 @@ import { toDate } from "../toDate/index.js";
 
 /**
  * @name addBusinessDays
- * @category Date Extension Helpers
+ * @category Day Helpers
  * @summary Add the specified number of business days (mon - fri) to the given date.
  *
  * @description
