diff --git a/source/vendor/supports-color/browser.js b/source/vendor/supports-color/browser.js
index f4e9acd9..4e75e86d 100644
--- a/source/vendor/supports-color/browser.js
+++ b/source/vendor/supports-color/browser.js
@@ -1,6 +1,8 @@
 /* eslint-env browser */
 
-const isBlinkBasedBrowser = /\b(Chrome|Chromium)\//.test(navigator.userAgent);
+const isBlinkBasedBrowser = navigator.userAgentData
+	? navigator.userAgentData.brands.some(({brand}) => brand === 'Chromium')
+	: /\b(Chrome|Chromium)\//.test(navigator.userAgent);
 
 const colorSupport = isBlinkBasedBrowser ? {
 	level: 1,
